tools/serverpod_cli/test/analyzer/entities/entity_analyzer/relation_manual_field_test.dart

```diff
@@ -0,0 +1,342 @@
+  child: List<ExampleChild>?, relation(field=myChildId)
+''',
+      Uri(path: 'lib/src/protocol/example.yaml'),
+      [],
+    );
+
+    var protocol2 = ProtocolSource(
+      '''
+class: ExampleChild
+table: example_child
+fields:
+  name: String
+  exampleId: int, relation(parent=example)
+''',
+      Uri(path: 'lib/src/protocol/example.yaml'),
+      [],
+    );
+
+    var definition1 = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol1,
+    );
+
+    var definition2 = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol2,
+    );
+
+    var entities = [definition1!, definition2!];
+    SerializableEntityAnalyzer.resolveEntityDependencies(entities);
+
+    SerializableEntityAnalyzer.validateYamlDefinition(
+      protocol1.yaml,
+      protocol1.yamlSourceUri.path,
+      collector,
+      definition1,
+      entities,
+    );
+
+    var errors = collector.errors;
+
+    test('then an error was collected.', () {
+      expect(errors, isNotEmpty);
+    });
+
+    test(
+        'then the error message reports that the field keyword cannot be used on a List relation.',
+        () {
+      var error = collector.errors.first;
+      expect(
+        error.message,
+        'The "field" property can only be used on an object relation.',
+      );
+    }, skip: errors.isEmpty);
+
+    test('then the error is reported at the field key location.', () {
+      var span = collector.errors.first.span;
+
+      expect(span?.start.line, 4);
+      expect(span?.start.column, 39);
+
+      expect(span?.end.line, 4);
+      expect(span?.end.column, 39 + 'field'.length);
+    }, skip: errors.isEmpty);
+  });
+
+  group('Given a class with an id relation with a field pointer defined', () {
+    var collector = CodeGenerationCollector();
+
+    var protocol1 = ProtocolSource(
+      '''
+class: Example
+table: example
+fields:
+  otherId: int
+  exampleChildId: int, relation(parent=example_child, field=otherId)
+''',
+      Uri(path: 'lib/src/protocol/example.yaml'),
+      [],
+    );
+
+    var protocol2 = ProtocolSource(
+      '''
+class: ExampleChild
+table: example_child
+fields:
+  name: String
+''',
+      Uri(path: 'lib/src/protocol/example.yaml'),
+      [],
+    );
+
+    var definition1 = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol1,
+    );
+
+    var definition2 = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol2,
+    );
+
+    var entities = [definition1!, definition2!];
+    SerializableEntityAnalyzer.resolveEntityDependencies(entities);
+
+    SerializableEntityAnalyzer.validateYamlDefinition(
+      protocol1.yaml,
+      protocol1.yamlSourceUri.path,
+      collector,
+      definition1,
+      entities,
+    );
+
+    var errors = collector.errors;
+
+    test('then an error was collected.', () {
+      expect(errors, isNotEmpty);
+    });
+
+    test(
+        'then the error message reports that the field keyword cannot be used on a List relation.',
```

Comment:
Question: This test message does't seem right since we are not working with lists. This is an id field direct relation right?

Reply:
correct, copy paste error.